wordpress-mobile/WordPress-iOS
# [Xcode 12] Fix Core Data transformer warnings

Fixes #15167. This is a new version of #15212, which adds a new core data model version, and fixes a crash on iOS 12. I've also added two migration tests. The remainder of this description is the same as the original PR:

This PR makes some changes to remove the is using a nil or insecure value transformer warnings when building with Xcode 12.

 - Updates all Transformable Core Data properties that use 'top level' classes to specify the NSSecureUnarchiveFromData transformer. This only exists on iOS 12+, so this will generate console warnings on iOS 11. However, if the class isn't found, I'm pretty certain that the system will just fallback to the default behaviour.
- Adds custom value transformers for the three Transformables that don't use top level classes – `Coordinate`, `NSError` (which does conform to NSSecureCoding), and `Set`.

**To test**

- Install a build that uses the 103 version of the data model
- In that build, create a post and in Post Settings assign it a location
- For a blog, go to Site Settings > Discussion, and add some entries to Hold For Moderation
- Also, begin uploading some media in the Media library and cause it to error - the easiest way is to disable your internet connection during the upload
- Background the app to ensure everything is saved
- Install a build from this branch, and ensure there are no Core Data errors logged on startup (you will initially see some from the pre-migration version of the database)
- Ensure that your post and its location are still correct
- Ensure that your site's Hold for Moderation settings have persisted
- Visit the media library, tap on one of your failed media items and check the error message is shown as the title of the alert controller that appears.
- Test with the Swift and the obj-c stack (you can disable the feature flag for this)

PR submission checklist:
